lib/Driver/ToolChains.cpp

```diff
@@ -847,6 +848,74 @@ toolchains::Darwin::constructInvocation(const InterpretJobAction &job,
   return II;
 }
 
+static StringRef
+getDarwinLibraryNameSuffixForTriple(const llvm::Triple &triple) {
+  switch(getDarwinPlatformKind(triple)) {
+    case DarwinPlatformKind::MacOS:
+      return "osx";
+    case DarwinPlatformKind::IPhoneOS:
+      return "ios";
+    case DarwinPlatformKind::IPhoneOSSimulator:
+      return "iossim";
+    case DarwinPlatformKind::TvOS:
+      return "tvos";
+    case DarwinPlatformKind::TvOSSimulator:
+      return "tvossim";
+    case DarwinPlatformKind::WatchOS:
+      return "watchos";
+    case DarwinPlatformKind::WatchOSSimulator:
+      return "watchossim";
+  }
+  llvm_unreachable("Unsupported Darwin platform");
+}
+
+static void AddLinkRuntimeLib(const ArgList &Args, ArgStringList &Arguments,
```

Comment:
Since this implementation will not work on non-Darwin, please put "Darwin" in the name.